lib/ansible/module_utils/k8s/raw.py

```diff
@@ -437,6 +436,9 @@ def _custom_condition(resource):
             # There should never be more than one condition of a specific type
             match = match[0]
             if match.status == 'Unknown':
+                if match.status == condition['status']:
+                    if condition.get('reason'):
+                        return match.reason == condition['reason']
```

Comment:
Should we also return outside of this conditional, in case they didn't specify a `reason` but did specify a `status` of `Unknown`?

Reply:
Or change to `condition.get('reason', True)` ?

Reply:
Could also be more explicit with `condition.get('reason') or 'reason' not in condition`